arbeitszeit_flask/api/input_documentation.py

```diff
@@ -1,54 +1,78 @@
-from typing import Callable, Sequence
+from typing import Any, Callable, Dict, Sequence, Union
 
 from flask_restx import Namespace
-from flask_restx.reqparse import RequestParser
 
 from arbeitszeit_web.api.controllers.parameters import (
-    FormParameter,
+    BodyParameter,
     PathParameter,
     QueryParameter,
 )
 
+Parameter = Union[QueryParameter, BodyParameter, PathParameter]
+TypeLookupTable = Dict[type, str]
+
+type_lookup_table: TypeLookupTable = {
+    str: "string",
+    int: "integer",
+    float: "number",
+    bool: "boolean",
+}
+
 
 class with_input_documentation:
     def __init__(
         self,
-        expected_inputs: Sequence[QueryParameter | FormParameter | PathParameter],
+        expected_inputs: Sequence[Parameter],
         namespace: Namespace,
     ) -> None:
         self._expected_inputs = expected_inputs
         self._namespace = namespace
 
     def __call__(self, func: Callable) -> Callable:
-        parser = RequestParser()
+        """See https://swagger.io/specification/v2 for more information on the OpenAPI specification."""
+        params: Dict[str, Any] = {}
         for expected_input in self._expected_inputs:
             if isinstance(expected_input, PathParameter):
-                # RequestParser does not handle path parameters
-                # https://github.com/noirbizarre/flask-restplus/issues/146
-                self._add_documentation_for_path_variable(func, expected_input)
+                self._add_path_parameter(params, expected_input)
```

Comment:
I am not sure if a separate function for each type is really required/helpful here (especially because the function is just an insert into a dict). I thought one advantage of those Unions would be that you could match them, so why not do this and then directly assign as done in the single funcs? Or, maybe have on "add" func that takes a "Parameter" type parameter and then does the type check?

Reply:
Consider the last commit I just pushed. I am not sure it it is better now, because we have now a single method that is quite long. But I have no strong opinions on this case, so if you prefer it that why, I am happy to merge it.
Let me also know, if I misunderstood your proposel.

Reply:
this is exactly what I meant. To me it is easier to read and extend, but maybe this is really personal preference. I think this is great work so I am going to approve the PR.